controllers/operatorpolicy_controller.go

```diff
@@ -332,6 +346,74 @@ func (r *OperatorPolicyReconciler) buildResources(policy *policyv1beta1.Operator
 	return sub, opGroup, updateStatus(policy, validationCond(validationErrors)), nil
 }
 
+// applySubscriptionDefaults will set the subscription channel, source, and sourceNamespace when they are unset by
+// utilizing the PackageManifest API.
+func (r *OperatorPolicyReconciler) applySubscriptionDefaults(
+	ctx context.Context, subscription *operatorv1alpha1.Subscription,
+) error {
+	subSpec := subscription.Spec
+
+	defaultsNeeded := subSpec.Channel == "" || subSpec.CatalogSource == "" || subSpec.CatalogSourceNamespace == ""
+
+	if !defaultsNeeded {
+		return nil
+	}
+
+	// PackageManifests come from an API server and not a Kubernetes resource, so the DynamicWatcher can't be used since
```

Comment:
Oh! Is that why you were saying the packagemanifest is inconsistent when you have multiple catalogsources for the same package? At least the resource isn't constantly changing in etcd I guess...

Reply:
Yes, exactly!